src/SourceBuild/content/test/Microsoft.DotNet.SourceBuild.SmokeTests/ArtifactsSize.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Formats.Tar;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.DotNet.SourceBuild.SmokeTests;
+
+[Trait("Category", "SdkContent")]
+public class ArtifactsSize : SmokeTests
+{
+    private readonly string baselineFilePath = BaselineHelper.GetBaselineFilePath($"ArtifactsSizes/{Config.TargetRid}.txt");
+    private readonly string[] baselineFileContent;
+    private readonly Regex buildVersionPattern = new(@"\b\d+\.\d+\.\d+[-@](alpha|preview|rc|rtm)\.\d(\.\d+\.\d+)?\b");
+
+
+    public ArtifactsSize(ITestOutputHelper outputHelper) : base(outputHelper)
+    {
+        baselineFileContent = File.Exists(baselineFilePath) ? File.ReadAllLines(baselineFilePath) : Array.Empty<string>();
+    }
+
+    [SkippableFact(new[] { Config.SourceBuiltArtifactsPathEnv, Config.SdkTarballPathEnv, Config.TargetRidEnv }, skipOnNullOrWhiteSpace: true)]
+    public void ArtifactsSizeTest()
+    {
+        Assert.True(Directory.Exists(BaselineHelper.GetBaselineFilePath("ArtifactsSizes/")));
+        Assert.NotNull(Config.SourceBuiltArtifactsPath);
+        Assert.NotNull(Config.SdkTarballPath);
+        Assert.NotNull(Config.TargetRid);
+
+        IEnumerable<TarEntry> artifactsTarEntries = Utilities.GetTarballContent(Config.SourceBuiltArtifactsPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+        IEnumerable<TarEntry> sdkTarEntries = Utilities.GetTarballContent(Config.SdkTarballPath).Where(entry => entry.EntryType == TarEntryType.RegularFile);
+
+        string[] tarEntries = sdkTarEntries.Concat(artifactsTarEntries)
+            .Select(entry =>
+            {
+                string modifiedPath = buildVersionPattern.Replace(entry.Name, "VERSION");
+                string result = modifiedPath.Replace(Config.TargetRid, "TARGET_RID");
+                return $"{result}: {entry.Length} bytes";
+            })
+            .OrderBy(entry => entry)
+            .ToArray();
+
+        foreach (string entry in tarEntries)
+        {
+            string tarEntryFilename = entry.Substring(0, entry.IndexOf(":")).Trim();
+            string baselineEntry = baselineFileContent?.FirstOrDefault(baselineEntry => baselineEntry.StartsWith(tarEntryFilename)) ?? "";
+
+            if (string.IsNullOrEmpty(baselineEntry))
+            {
+                LogWarningMessage($"{tarEntryFilename} does not exist in baseline. Adding it to the baseline file");
+                File.AppendAllText(baselineFilePath, $"{entry}" + Environment.NewLine);
+            }
+            else
+            {
+                string tarEntrySize = entry.Substring(entry.IndexOf(":") + 1).Replace(" bytes", "").Trim();
+                string baselineEntrySize = baselineEntry.Substring(tarEntryFilename.Length + 1).Replace(" bytes", "").Trim() ?? "0";
```

Comment:
I did consider using JSON but ultimately decided to go with the current implementation because the extra information needed in a JSON felt repetitive and impacted the readability.

```
{
"centos-8-x64":[
    {"FileName":"file1", "Bytes":"1"},
    {"FileName":"file2", "Bytes":"2"},
    {"FileName":"file3", "Bytes":"1"},
    .....
],
"centos-9-x64":[
    {"FileName":"file1", "Bytes":"1"},
    {"FileName":"file2", "Bytes":"2"},
    {"FileName":"file3", "Bytes":"1"},
    .....
],
...
}
```

In my opinion, using a JSON meant implementing in one of two ways:
1) One large JSON baseline with information from every target RID -> extremely large file that is difficult to parse through and could lead to issues when looking for a specific filename/RID combination since many filenames are repeated across RIDs.
2) Multiple large JSON baselines, each with information from a single target RID -> information in the line (such as "FileName", "Bytes", '{}', etc." that impact readability.

I can see a JSON being extremely helpful if we had a tool that parsed the information and displayed it with a clean UI (i.e. dropdowns for each RID), but I think that would involve extra tools that I am either not aware of or that we do not have.